### ISSUE TITLE
Validate report 'magic header not detected'

### ISSUE THREAD

**Issue by wangshishuo** [second-state/WasmEdge-go#14]
If I load wasm using vm.LoadWasmBuffer, then the vm.Validate() will result with an error: 'magic header not detected'

**Comment by hydai** [comment]
Could you also provide the wasm file?

**Comment by wangshishuo** [comment]
[https://github.com/second-state/WasmEdge-go-examples/tree/master/go_ExternRef](https://github.com/second-state/WasmEdge-go-examples/tree/master/go_ExternRef)

Only change 

``` go
vm.LoadWasmFile(os.Args[1])
```

to

```go
b, er := ioutil.ReadFile(os.Args[1])
  if er != nil {
    log.Fatal(er)
  }
  er = vm.LoadWasmBuffer(b)
  if er != nil {
    log.Fatal(er)
  }
```

You will get the error.

**Comment by q82419** [comment]
This is fixed in the v0.9.0-rc1.
